lib/bookings_repository.py

```diff
@@ -23,8 +27,14 @@ def find_available_spaces(self, month):
             return spaces
         else:
             spaces = []
-            rows = self._connection.execute('SELECT DISTINCT (spaces.id), spaces.title, spaces.description, spaces.email_address, spaces.price_per_night, spaces.user_id FROM spaces FULL OUTER JOIN bookings ON spaces.id = bookings.spaces_id WHERE spaces.id != %s ORDER BY spaces.id', [space_id])
+            available_spaces_list = self._connection.execute('SELECT spaces.id, spaces.title, spaces.description, spaces.email_address, spaces.price_per_night, spaces.user_id FROM spaces WHERE spaces.id NOT IN (SELECT DISTINCT spaces_id FROM bookings WHERE booked_month = %s)', [month])
+            for row in available_spaces_list:
+                item = Spaces(row['id'], row['title'], row['description'], row['email_address'], row['price_per_night'], row['user_id'])
+                spaces.append(item)
+            return spaces
+            
+            """rows = self._connection.execute('SELECT DISTINCT (spaces.id), spaces.title, spaces.description, spaces.email_address, spaces.price_per_night, spaces.user_id FROM spaces FULL OUTER JOIN bookings ON spaces.id = bookings.spaces_id WHERE spaces.id != %s ORDER BY spaces.id', [space_id])
```

Comment:
I would suggest to comment what this commented out code is for so we can come back to it if needed